coldfront/core/project/management/commands/add_service_units_to_project.py

```diff
@@ -0,0 +1,154 @@
+import logging
+from decimal import Decimal
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.management import BaseCommand, CommandError
+
+from coldfront.config import settings
+from coldfront.core.project.models import Project
+from coldfront.core.statistics.models import ProjectTransaction
+from coldfront.core.statistics.models import ProjectUserTransaction
+from coldfront.core.utils.common import utc_now_offset_aware
+from coldfront.api.statistics.utils import get_accounting_allocation_objects
+from coldfront.api.statistics.utils import set_project_allocation_value
+from coldfront.api.statistics.utils import set_project_user_allocation_value
+from coldfront.core.allocation.models import AllocationAttributeType, Allocation
+
+
+class Command(BaseCommand):
+    help = 'Command to add SUs to a given project.'
+    logger = logging.getLogger(__name__)
+
+    def add_arguments(self, parser):
+        parser.add_argument('--project_name',
+                            help='Name of project to add SUs to.',
+                            type=str,
+                            required=True)
+        parser.add_argument('--amount',
+                            help='Number of SUs to add to a given project.',
+                            type=int,
+                            required=True)
+        parser.add_argument('--reason',
+                            help='User given reason for adding SUs.',
+                            type=str,
+                            required=True)
+        parser.add_argument('--dry_run',
+                            help='Display updates without performing them.',
+                            action='store_true')
+
+    def validate_inputs(self, options):
```

Comment:
DB calls / calculations are getting repeated; you could return the cleaned values in `validate_inputs` and use them in `handle`.